assets/js/googlesitekit/widgets/util/get-widget-component-props.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Widgets component props utilities.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import WidgetReportZero from '../components/WidgetReportZero';
+import WidgetActivateModuleCTA from '../components/WidgetActivateModuleCTA';
+import WidgetCompleteModuleActivationCTA from '../components/WidgetCompleteModuleActivationCTA';
+
+/**
+ * Gets the props to pass to a widget's component.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} widgetSlug The widget's slug.
+ * @return {Object} Props to pass to the widget component.
+ */
+export function getWidgetComponentProps( widgetSlug ) {
+	// Scope widget-specific components to the widget instance so that the
+	// component does not need to (re-)specify the widget slug.
+	return {
+		WidgetReportZero: withWidgetSlug( widgetSlug )( WidgetReportZero ),
+		WidgetActivateModuleCTA: withWidgetSlug( widgetSlug )( WidgetActivateModuleCTA ),
+		WidgetCompleteModuleActivationCTA: withWidgetSlug( widgetSlug )( WidgetCompleteModuleActivationCTA ),
+	};
+}
+
+function withWidgetSlug( widgetSlug ) {
+	return ( WrappedComponent ) => {
+		return ( props ) => <WrappedComponent widgetSlug={ widgetSlug } { ...props } />;
```

Comment:
Should we not pass `widgetSlug` after the props spread here? My original thought was that the scoped components would not be able to pass a different slug (or at least it wouldn't make a difference if it did). I can't think of a reason why a widget would want/need to use a different slug for these components. Since it's intended to only be used within that widget, I think it should be fixed to only work for that widget.
```suggestion
		return ( props ) => <WrappedComponent { ...props } widgetSlug={ widgetSlug } />;
```